integration-tests/pkg/mock_sensor/server.go

```diff
@@ -155,24 +155,28 @@ func (m *MockSensor) Connections(containerID string) []types.NetworkInfo {
 	defer m.networkMutex.Unlock()
 
 	if connections, ok := m.connections[containerID]; ok {
-		keys := make([]types.NetworkInfo, 0, len(connections))
-		for k := range connections {
-			keys = append(keys, k)
-		}
-		return keys
+		return connections
```

Comment:
I think this is unsafe, we're returning a reference to the connections slice, and that slice may be modified by both the sensor thread (more connections come in for this containerID) and the main thread (sorting, asserting, etc)

```suggestion
        conns := make([]types.NetworkInfo, len(connections))
        copy(conns, connections)
		return conns
```